src/org/python/core/PyByteArray.java

```diff
@@ -1335,17 +1326,14 @@ public PyObject __iadd__(PyObject o) {
 
     @ExposedMethod(type = MethodType.BINARY, doc = BuiltinDocs.bytearray___iadd___doc)
     final synchronized PyObject bytearray___iadd__(PyObject o) {
-        PyType oType = o.getType();
-        if (oType == TYPE) {
-            // Use the general method, specifying the crack at the end of the array.
-            // Note this deals with the case o==this.
-            setslice(size, size, 1, (BaseBytes)o);
-        } else if (oType == PyString.TYPE) {
-            // Will fail if somehow not 8-bit clean
-            setslice(size, size, 1, (PyString)o);
+        if (o instanceof BufferProtocol && !(o instanceof PyUnicode)) {
```

Comment:
You just need `if setsliceFromBuffer(size, size, 1, o) { }` as the third possibility, with a comment so it doesn't l;ook like a typo. `setsliceFromBuffer` itself checks for `BufferProtocol` and `PyUnicode` will reject you taking the buffer. Compare existing use in `setslice(int,int,int,PyObject)`.